io/www/src/fragment/pipeline.js

```diff
@@ -36,19 +52,38 @@ async function main(params) {
     if (!context.state) {
         context.state = await stateLib.init();
     }
-    context.debugLogs = (await context.state.get('debugFragmentLogs'))?.value || false;
-    const requestKey = `req-${context.id}-${context.locale}`;
-    const cachedMetadataStr = (await context.state.get(requestKey))?.value;
-    let cachedMetadata = {};
-    if (cachedMetadataStr) {
-        log(`(${Date.now() - startTime}ms) found cached metadata for ${requestKey} -> ${cachedMetadataStr}`, context);
-        try {
-            cachedMetadata = JSON.parse(cachedMetadataStr);
-            const { translatedId, dictionaryId } = cachedMetadata || {};
-            context = { ...context, translatedId, dictionaryId };
-        } catch (e) {
-            logError(`Error parsing cached metadata ${cachedMetadataStr}: ${e.message}`, context);
+    try {
+        const { json } = await getJsonFromState('network-config', context);
+        context.networkConfig = json || {};
+        const timeout = context.networkConfig.mainTimeout || 5000;
+        return await Promise.race([mainProcess(context), createTimeoutPromise(timeout)]);
```

Comment:
what i observe is that its not easy to differentiate fetch timeout from overall pipeline timeout